client/src/components/History/Content/Collection/CollectionProgress.vue

```diff
@@ -3,11 +3,15 @@ at components/JobStates/CollectionJobStates but it relies on the backbone data
 model, so probably has to go eventually.-->
 <template>
     <div class="collection-progress">
-        <b-progress v-if="maxJobs && runningJobs" :max="maxJobs" show-value>
-            <b-progress-bar v-if="errorJobs" :value="errorJobs" variant="danger" />
-            <b-progress-bar v-if="okJobs" :value="okJobs" variant="success" />
-            <b-progress-bar v-if="runningJobs" :value="runningJobs" variant="warning" />
-            <b-progress-bar v-if="waitingJobs" :value="waitingJobs" variant="secondary" />
+        <b-progress v-if="maxJobs != okJobs + errorJobs" :max="maxJobs">
```

Comment:
You have a terminal job abstraction in the `JobStateSummary` class, can you use that please ?

Reply:
That abstraction layer needs to be revisited, it remained from the previous code but is not fully functional nor used.

Reply:
Well, the solution cannot possibly be to define things ad-hoc wherever you need it, that way we end up with the subtle bugs that you and assunta are fixing now. 'Cause here you're missing the deleted state, and that's correctly defined in the `JobStateSummary` class. Moreover assunta is already using this in this component.

Reply:
I am suggesting to cleanly use `models/state.js`. The `JobStateSummary` is rarely used right now, its usage was mostly reintroduced with the progress bar.

Reply:
I tried using the terminal job abstraction in JobStateSummary, but I'm facing difficulty making it reactive, so it's not really viable right now.

Reply:
Found a major problem with JobStateSummary (the states in that class don't actually match up to the states in state.js), after fixing that, this wasn't hard at all! 😅